File: app/src/main/java/sk/solver/weatherapp/WeatherClient.java
package sk.solver.weatherapp;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import sk.solver.weatherapp.models.WeatherResponse;

public interface WeatherClient {

    @GET("weather")
    Observable<WeatherResponse> getWeather(@Query("q") String cities, @Query("units") String units, @Query("appid") String appId);
}

File: app/src/main/java/sk/solver/weatherapp/WeatherClient.kt
package sk.solver.weatherapp

import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import sk.solver.weatherapp.models.WeatherResponse

interface WeatherClient {

    @GET("weather")
    fun getWeather(
        @Query("q") cities: String?,
        @Query("units") units: String?,
        @Query("appid") appId: String?
    ): Observable<WeatherResponse>?
}